src/Euro.php

```diff
@@ -105,6 +113,10 @@ public static function newFromFloat( float $euroAmount ): self {
 	 * @throws InvalidArgumentException
 	 */
 	public static function newFromInt( int $euroAmount ): self {
+		if ( $euroAmount > floor( PHP_INT_MAX / 101 ) ) {
```

Comment:
why divide by 101?
The best I could find after googling is [the wikipedia article on palindromic primes](https://en.wikipedia.org/wiki/Palindromic_prime)

Reply:
I expected the /100 to work but it does not. Hence 101. That likely does not allow representing certain big numbers that strictly speaking can be represented though that seems like something that does not actually matter here. There might also be some small range of numbers for which the type error still happens, though again these will be very large ones that don't really matter. I expect this maybe-not-100%-correct-but-definitely-more-correct-than-before change to take care of all errors that currently end up in the log.